### DISCUSSION TITLE
画像を連打で消す

### DISCUSSION THREAD

**Discussion by ShunHigu** [lit-iphone-course/lit-iphone-course#76] [Question-original]
１　画像を連打することでその画像が消えるコードを書きたいです。

２　画像を連打したいのですが、エラー文を見ているとfor文に間違いがあると考えます。
Missing return in instance method expected to return 'UIImageView'
自分は何の画像が連打されているかを明確にするためにfor文を用いているのですが、for文を使う以外の方法がわからないので教えて頂きたいです。

３　連打文の参考サイトです
https://blog.ch3cooh.jp/entry/20150629/1435541400

４　var e:Int!
    func getView() -> UIImageView {
        for e in 0...enemynum{
            // ダブルタップ
            let doubleTap = UITapGestureRecognizer(target: EnemyImage, action: "tappdDouble:")
            doubleTap.numberOfTapsRequired = 2
            EnemyImage[e].addGestureRecognizer(doubleTap)
        }
    }
    //連続タップで敵の画像が消える
    func tappdDouble(sender: UITapGestureRecognizer!) {
        // ダブルタップしたときの処理
        EnemyImage[e].isHidden=true
    }

５　https://github.com/ShunHigu/OriginalNew

**Comment by Yonpachi48** [comment]
ダブルタップで敵を攻撃させる処理を書いてるけど、連打だったらダブルタップとはまた違うと思います！
一番簡単なのは敵がタップされた時に敵のHPを減らす処理と、HPが0になったら敵(敵の画像)を消せばいいんじゃないかな？？

**Reply by ShunHigu** [reply]
なるほど！ダブルタップの2の数字を6などに変えるつもりでしたが、四ぱちさんのいうやり方の方が効率的そうなのでやってみます！

**Reply by ShunHigu** [reply]
敵をタップする際に、画面上に敵が複数いる際はどのようにタップされた画像が何かを判別すれば良いのでしょうか？

**Comment by txxxxc** [comment]
`Missing return in instance method expected to return 'UIImageView'`の部分は文法エラーなのでもう一度見返して実装してもらいたいです！

**Reply by ShunHigu** [reply]
配列にviewの情報を入れている場合に、タップしたviewが配列の何番目かを判断するコードが見つからないので教えていただきたいです！

**Reply by ShunHigu** [reply]
for文で全部を読み込ませるというやり方だと絶対にエラーが出てしまうんです...
